snap/snapcraft.yaml

```
@@ -35,12 +35,12 @@ apps:
     environment: &_environment
       # Prep for Mir
       MIR_SERVER_PLATFORM_PATH: ${SNAP}/usr/lib/${CRAFT_ARCH_TRIPLET_BUILD_FOR}/mir/server-platform
-      __EGL_VENDOR_LIBRARY_DIRS: $SNAP/etc/glvnd/egl_vendor.d:$SNAP/usr/share/glvnd/egl_vendor.d
+      __EGL_VENDOR_LIBRARY_DIRS: /var/lib/snapd/lib/glvnd/egl_vendor.d:$SNAP/etc/glvnd/egl_vendor.d:$SNAP/usr/share/glvnd/egl_vendor.d
       LIBGL_DRIVERS_PATH: $SNAP/usr/lib/${CRAFT_ARCH_TRIPLET_BUILD_FOR}/dri
       LIBINPUT_QUIRKS_DIR: ${SNAP}/usr/share/libinput
       PATH: ${SNAP}/bin/:${SNAP}/usr/bin/:${SNAP}/usr/local/bin/:${PATH}
       # For "reasons" this is being set despite this being a classic snap. This overwrites the nonsense
-      LD_LIBRARY_PATH: ""
+      LD_LIBRARY_PATH: "/var/lib/snapd/lib/gl"
```

This shouldn't be needed, as it will search system locations anyway.

```suggestion
      LD_LIBRARY_PATH: ""
```